lib/straight/exchange_rate_adapters/average_rate_adapter.rb

```diff
@@ -0,0 +1,22 @@
+module Straight
+  module ExchangeRate
+
+    class AverageRateAdapter
+
+      # Takes exchange rate adapters instances as an arguments
+      def initialize(*adapter_instances)
+        @rates = []
+        adapter_instances.each do |adapter|
+          @rates << adapter.rate_for('USD')
+        end
+      end
+
+      def calculate_avg_rate
```

Comment:
This method should be called the same as in other adaters: `#rate_for`. Otherwise, I'd have to change other parts of Straight to incorporate your adapter, which is definitely not something that should ever be done in such a case. This is why we have an idea of "polymorphism": I can use any child class that inherits from `Straight::ExchangeRate::Adapter` and expect to find a `#rate_for` method there.